src/System.Collections/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -247,9 +247,9 @@ private void CopyTo(KeyValuePair<TKey, TValue>[] array, int index)
                 throw new ArgumentNullException("array");
             }
 
-            if (index < 0 || index > array.Length)
+            if ((uint)index > (uint)array.Length)
```

Comment:
I suggested it as a possibility at https://github.com/dotnet/corefx/pull/5672#discussion_r51124178
If someone finds it unclear I think it shouldn't be done. (Its become an idiom to me, so I can't really judge how clear or unclear it is).